conductor_api/src/conductor/base.rs

```diff
@@ -155,8 +153,13 @@ pub fn notify(msg: String) {
 
 impl Conductor {
     pub fn from_config(config: Configuration) -> Self {
-        let rules = config.logger.rules.clone();
         lib3h_sodium::check_init();
+        let _rules = config.logger.rules.clone();
+        let logger = FastLoggerBuilder::new()
+            .set_level_from_str("Trace")
+            .add_rule_filter(RuleFilter::new("Abort", false, "Red"))
+            .build()
+            .expect("Fail to instanciate the logging factory.");
```

Comment:
@dymayday, I've moved the initialization here because this is where we get the conductor config struct (the parameter to this function). This still leaves us with this panic:
```
!C 'alice'! thread 'jsonrpc-eventloop-0' panicked at 'Fail to send message to the logging thread.: "SendError(..)"', src/libcore/result.rs:1051:5

!C 'alice'! thread 'jsonrpc-eventloop-0' panicked at 'Fail to send message to the logging thread.: "SendError(..)"', src/libcore/result.rs:1051:5
stack backtrace:

!C 'alice'!    0:     0x564e0c8a441b - backtrace::backtrace::libunwind::trace::hd4effc5cbc99af65
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1:     0x564e0c8a441b - backtrace::backtrace::trace_unsynchronized::h389cccb5967fa3c2
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2:     0x564e0c8a441b - std::sys_common::backtrace::_print::hbb606689fb98322f
                               at src/libstd/sys_common/backtrace.rs:47
   3:     0x564e0c8a441b - std::sys_common::backtrace::print::h2b6171d791ff1fac
                               at src/libstd/sys_common/backtrace.rs:36
   4:     0x564e0c8a441b - std::panicking::default_hook::{{closure}}::h7c4f6f6170cc3cf6
                               at src/libstd/panicking.rs:200
   5:     0x564e0c8a40f7 - std::panicking::default_hook::hd9f23e1c4cbbedcb
                               at src/libstd/panicking.rs:214
   6:     0x564e0c8a4b90 - std::panicking::rust_panic_with_hook::h7fedb25d8f7a616a
                               at src/libstd/panicking.rs:477
   7:     0x564e0c8a4712 - std::panicking::continue_panic_fmt::hed2a79ffb3c5014c
                               at src/libstd/panicking.rs:384
   8:     0x564e0c8a45f6 - rust_begin_unwind
                               at src/libstd/panicking.rs:311

!C 'alice'!    9:     0x564e0c8c782d - core::panicking::panic_fmt::hcd868b6ef07f5b76
                               at src/libcore/panicking.rs:85
  10:     0x564e0c180a92 - core::result::unwrap_failed::h2b0833d5f0b333f3
  11:     0x564e0c1726ac - <logging::FastLogger as log::Log>::log::hdde04fc0a6ac2298
  12:     0x564e0c74bd86 - log::__private_api_log::h0ecd678ad78f5d89
  13:     0x564e0c6bb329 - <tokio_threadpool::worker::Worker as core::ops::drop::Drop>::drop::h046f61df1fe281dd
  14:     0x564e0c6b940f - core::ptr::real_drop_in_place::h17182bfcf38320bb
  15:     0x564e0c6b9199 - std::sys_common::backtrace::__rust_begin_short_backtrace::h5fad3be99810a963
  16:     0x564e0c6be927 - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h5b77da233c2548f6
  17:     0x564e0c6bb71d - std::panicking::try::do_call::hdab2476811739eb9

!C 'alice'!   18:     0x564e0c8aed8a - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:82
  19:     0x564e0c6bb64b - std::panicking::try::h7775786e79e4ae5a
  20:     0x564e0c6be9b1 - std::panic::catch_unwind::ha1c48153e24694a0
  21:     0x564e0c6be48b - core::ops::function::FnOnce::call_once{{vtable.shim}}::h20aa994d92e17cd2
  22:     0x564e0c89238f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h56322f6b657cd33d
                               at /rustc/69656fa4cbafc378fd63f9186d93b0df3cdd9320/src/liballoc/boxed.rs:766
  23:     0x564e0c8ae140 - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h2cbeedc59f1a5bf0
                               at /rustc/69656fa4cbafc378fd63f9186d93b0df3cdd9320/src/liballoc/boxed.rs:766
  24:     0x564e0c8ae140 - std::sys_common::thread::start_thread::h780c24dc0e8bc388
                               at src/libstd/sys_common/thread.rs:13
  25:     0x564e0c8ae140thread panicked while panicking. aborting.
```

As a first try I had it in `conductor/src/main.rs` but the problem is that we need to initialize the logger with the info that is in the `config` we get here. Can we make so that those first logs that occur before the logger is initialized to pick the messages up on the other end of the channel don't panic? For instance by using unbounded crossbeam channels?

Reply:
Correction: the same error shows up when the logger gets initialized in the conductor main function as seen here: https://github.com/holochain/holochain-rust/pull/1614